### PR TITLE
fix: align title and title content in Dynamic Page

### PR DESCRIPTION
**Pull request by kavya-b**
## Related Issue
Closes SAP/fundamental-styles#1800

## Description
This PR does the following:
- aligns title and title content in Dynamic Page 
- modifies z-index usage to fix look in mobile popover case
- keeps title content ellipsible and restricts to single line to maintain alignment
- uses variable for common value

Breaking change:
- removed modifier class `fd-dynamic-page__collapsible-header-container--collapsed`

### Before:
alignment: image
popover getting cut: image
title content multi-line: image

### After:
alignment: image
popover getting cut fixed: image
title-content single-line: image

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
4. Documentation
- [x] Storybook documentation has been created/updated
- [x] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

**Review by katekozlowska — CHANGES_REQUESTED** [review]
There should be truncation on longer text, exampe from IE11:

image

**Comment by kavya-b** [comment]
> There should be truncation on longer text, exampe from IE11:
> 
> image

unfortunately this is happening due to using `display: table-cell` . It expects a max-width value(but doesn't like % values) and I don't think max-width in rem should be provided when the content could be anything, and spec doesn't specify how much space title and title content should each take. 
something similar is happening with Wizard as well, @InnaAtanasova can you help a bit with this? Is this a known issue in IE where long text truncation in an element that is a `table-cell` has problems?

I am currently revisiting the earlier `display: flex` logic again; I remember it didn't align  well in some cases which is why I switched to `table-cell` in the first place 😓

**Comment by kavya-b** [comment]
@katekozlowska fixed in IE with `display:flex` usage. Please check.

**Review by salarenko — APPROVED** [review]

File: src/dynamic-page.scss [hunk 1]
```diff
@@ -272,6 +282,7 @@ $block: #{$fd-namespace}-dynamic-page;
 
   .#{$fd-namespace}-toolbar.#{$block}__title-content-toolbar {
```
Comment (InnaAtanasova) [inline src/dynamic-page.scss:283]: Can you remove the `.#{$fd-namespace}-toolbar` ? This is not making the component self-contained. If the rules in `.#{$block}__title-content-toolbar` are not taking precedence over the ones coming from ` .#{$fd-namespace}-toolbar` you should apply more specificity.

Comment (kavya-b) [reply src/dynamic-page.scss]: done, also removed mixed scope from breadcrumb and tabs.

**Review by katekozlowska — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Pull request merged**